src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -1024,14 +1028,16 @@ bool KeyCondition::transformConstantWithValidFunctions(
                             auto left_arg_type = left->result_type;
                             auto left_arg_value = (*left->column)[0];
                             std::tie(const_value, const_type) = applyBinaryFunctionForFieldOfUnknownType(
-                                func->function_builder, left_arg_type, left_arg_value, const_type, const_value);
+                                FunctionFactory::instance().get(func->function_base->getName(), context),
```

Comment:
Maybe better to change `applyBinaryFunctionForFieldOfUnknownType` to work with `IFunctionBase`.